### ISSUE TITLE
Restructure codebase and packages

### ISSUE THREAD

**Issue by srishanbhattarai** [srishanbhattarai/nepcal#33]
The initial design for the package structure was simple:
1. `dateconv` package would provide functions to convert to and from Nepali dates
2. `main` package would be the executable package that would render the calendar

While it has worked fine 'til now, as more and more "non-conversion" functionality gets added, the `dateconv` package doesn't make sense because the package does much more than just convert dates.

**Comment by srishanbhattarai** [comment]
#  Simple 2-folder structure
```
/cmd 
    /nepcal  <-- binary
        main.go
/nepcal <-- root package
    conv.go <-- conversion stuff
    time.go  <-- particular utility methods with the BS dates
```

The `cmd/` folder would just be the current main package and the `nepcal` package would be the brains of it all.

## `nepcal` package and relevant changes
* Model it after the `time` package.
* There will be a central `Time` struct off of which we attach methods for all utilities. This should be fine because everything will be namespaced by the package name `nepcal.Time`
  - This means that the public API needs to change
  - For e.g. [these](https://github.com/srishanbhattarai/nepcal/blob/master/dateconv/dateconv.go#L86-L98) functions now become part of the `Time` struct.
  - Consult the [methods](https://golang.org/pkg/time) that hang off `time.Time` for reference.
  - There can be a `time.Time` struct embedded inside `nepcal.Time`
* Any functions that aren't methods of the `Time` struct will (probably only) be constructors to create the structure.
  - For e.g the [time.Date](https://golang.org/src/time/time.go?s=41192:41275#L1350) function will similarly be implemented as `nepcal.Date` with the same API. The caller does the extra business of argparsing, which is already the case.
  - Similarly, the `nepcal.Unix` function can be a similar equivalent.
  -  This imposes certain limitations as we cannot have functions to ask the package "What is the BS equivalent weekday of Wednesday?". This is probably silly to ask anyway. The struct can still expose the weekday.
  - Change functions like [`totalDaysSpannedUntilDate`](https://github.com/srishanbhattarai/nepcal/blob/af09f92f2603c8b2ca8ce98bd115339fe8f98a6f/dateconv/dateconv.go#L138-L152) to be implemented using methods

**Comment by srishanbhattarai** [comment]
Fixed in [v1.0.0](https://github.com/srishanbhattarai/nepcal/releases/tag/v1.0.0).
